### ISSUE TITLE
Grouped Boxplot bar width broken with numeric X values

### ISSUE THREAD

**Issue by BioTurboNick** [JuliaPlots/StatsPlots.jl#234]
```
a = randn(6000);
b = repeat([1;2;3], 2000);
c = repeat([1;2], 3000);
groupedboxplot(b,a,group=c)
```
This is producing infinitesimally thin boxes. It appears to be due to line 135 in boxplot.jl:
`bar_width --> (0.8 * mean(diff(x)))`

`mean(diff(x))` evaluates to ~0.0003334.

It looks like it was copied from groupedbar, which just has one x value per column.

**Comment by BioTurboNick** [comment]
I believe `bar_width --> (0.8 * mean(diff(unique(x))))` achieves what was desired. I'll create a PR.

**Comment by mkborregaard** [comment]
Thanks for closing this (and the PR). Pro tip: add "closes #234" to the title of the PR to have it close the issue automatically when merged.
